### PR TITLE
Fix/GitHub vulnerabilities

### PR DESCRIPTION
**Pull request by 5uryan5h**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Areas Touched

- [ ] `apps/backend` (API / worker)
- [ ] `apps/dashboard`
- [ ] `apps/electron`
- [ ] `apps/xyne-claw` / `apps/xyne-claw-auth`
- [ ] `packages/shared` or another shared package
- [ ] Infra (docker, scripts, nix, CI)

---

## Zero (Sync) Changes

- [ ] No Zero changes — **skip this section**
- [ ] Adds/modifies a **mutator**
- [ ] Adds/modifies the **schema** (tables, columns, relationships)
- [ ] Adds/modifies a **query or ACL**

If any of the above:

- [ ] Server (`apps/backend/src/zero/mutators.ts`) and client (`apps/dashboard/src/zero/mutators.ts`) mutators mirror each other — same namespace, name, and args schema
- [ ] No `Date.now()` / `uuid()` generated inside a mutator
- [ ] Optimistic client result matches the server result (no state flash on rebase)
- [ ] New tables exported in `createSchema()` and given a Query ACL registered in `query-acl-factory.ts`
- [ ] Matching Prisma model + migration, client regenerated

## Schema & Data Changes

- [ ] No schema changes — **skip this section**
- [ ] Prisma schema modified (`apps/backend/prisma` and/or `prisma-common`)
- [ ] Migration added
- [ ] Backfill / data migration included

If any of the above:

- [ ] Clients regenerated (`db:generate`, `db:common:generate`)
- [ ] No new Postgres enum or enum value — enums are frozen (`pnpm run test:enum`)
- [ ] Backward compatible with deployed code, or rollout order noted below
- [ ] Backfill is idempotent

## Config, Environment & URLs

- [ ] No config changes — **skip this section**
- [ ] Env variable added / renamed / removed
- [ ] **Base URL, host, port, or origin changed** (API, Zero, Vespa, Claw, LiveKit, LiteLLM, …)
- [ ] CORS / allowed origins / OAuth redirect URIs changed
- [ ] Feature flag or runtime config changed

If any of the above:

- [ ] Key added to **all** relevant env files, not just the one you run — `apps/backend/.env.{example,local,test}`, `apps/dashboard/.env.{example,local,test}`, plus the app you touched
- [ ] Env setup / secret scripts and docker compose updated
- [ ] Verified in every consumer with its own base URL — dashboard (`VITE_API_BASE_URL`, `VITE_ZERO_SERVER`), Electron (`VITE_ELECTRON_*`), shareable links (`VITE_SHAREABLE_ORIGIN`), server-to-server (`BACKEND_URL`, `XYNE_CLAW_URL`, `XYNE_CLAW_AUTH_URL`)
- [ ] Google / Microsoft OAuth redirect URIs still resolve
- [ ] Nothing hardcoded, no secrets committed

## API Contract

- [ ] No API changes
- [ ] New endpoint
- [ ] Existing contract modified (shape, status codes, auth)
- [ ] Breaking for dashboard / electron / external / bots — migration noted above

---

### Automated

- [ ] Backend unit tests — `pnpm --filter xyne-spaces-backend run test`
- [ ] Claw tests — `pnpm --filter xyne-claw run test`
- [ ] End-to-end suite — `pnpm run test`
- [ ] Added / updated tests for this change
- [ ] Not covered by tests 

### Manual

**Users**
- [ ] Single user
- [ ] Two users at once (two accounts / two browsers) — sync lands on both, no cross-user leakage
- [ ] Different roles or permission levels (member vs admin, ACL boundaries)
- [ ] Different workspaces / spaces — data stays scoped

**Login**
- [ ] Google
- [ ] Microsoft
- [ ] Dev auth (`ENABLE_DEV_AUTH`)
- [ ] Fresh signup / first-time user
- [ ] Session expiry, re-login, logout

**Run mode**
- [ ] Local dev (`pnpm run dev:all`)
- [ ] Test mode (`dev:test`)
- [ ] Sandbox / claw test (`claw:test`)
- [ ] Prod-like local (`dev:prod`)
- [ ] Docker compose
- [ ] Electron desktop

**Sync & resilience** — for anything touching Zero
- [ ] Multiple tabs stay consistent
- [ ] Reload / hard refresh — no duplicate or lost rows
- [ ] Offline then reconnect — pending mutations replay cleanly
- [ ] Concurrent edits on the same entity by two users

**Surface & data**
- [ ] Web browser
- [ ] Electron desktop
- [ ] Narrow viewport, dark + light theme
- [ ] Empty state
- [ ] Large / realistic data volume
- [ ] Error paths (network failure, denied permission, invalid input)

---

## Checklist

- [ ] Reviewed my own diff; scoped to one thing
- [ ] `pnpm run build:shared` passes
- [ ] Backend `typecheck` + `build` pass
- [ ] Dashboard `lint:errors-only` + `typecheck` + `build` pass
- [ ] Formatting clean in the packages I touched
- [ ] New deps added with `pnpm --filter <pkg> add <dep>`; version pins in root `pnpm.overrides`
- [ ] Commits follow `<type>: <TICKET-ID> <subject>`; branch is `fix/*` or `feature/*`
- [ ] Docs / guidelines updated where behaviour changed
- [ ] No debug logging or commented-out code left behind

### CODE REVIEW

**Review by XyneSpaces — COMMENTED** [review]
Automated review — 3 blocker(s), 4 nit(s). **Changes requested.** Core concern: the cross-major dependency overrides (uuid@3/8/9/10 → 11.1.1 at root and in kata-sdk) can break transitive consumers that rely on uuid's removed callable default export, and the pi-coding-agent bump diverges from its pinned sibling packages. The one functional code change (PdfJsStrategy v6 destroy migration) is correct. Blockers: (1) uuid cross-major overrides, (2) kata-sdk uuid override vs request/k8s client, (3) pi-coding-agent version divergence / likely out-of-scope. Also: six-major Electron bump and nodemailer major bump both need real runtime validation, and the PR description/testing checklist is empty — fill it in and attach POT before merge.

File: package.json [hunk 1]
```diff
@@ -221,77 +220,96 @@
...
+      "react-router": "7.18.2",
+      "sharp": "0.35.0",
       "shell-quote": "1.10.0",
       "simple-git": "3.36.0",
       "socket.io-parser": "4.2.7",
       "styled-components>postcss": "8.5.10",
       "systeminformation": "5.31.17",
       "tar@6": "7.5.20",
-      "tar@7": "7.5.20",
+      "tar@7": "7.5.21",
```
Comment (XyneSpaces) [inline package.json:302]: Forcing `uuid@3`, `uuid@8`, `uuid@9`, and `uuid@10` all to `11.1.1` overrides transitive consumers across major versions. uuid v3 exposes a callable default export (`require('uuid')()`) that v7+ removed, so any transitive package using the old API will throw `uuid is not a function` at runtime. Scope these overrides to compatible ranges, or verify every transitive consumer only uses the named `v4`/`v1` API before forcing v11.

File: packages/kata-sdk/package.json [hunk 2]
```diff
@@ -20,6 +20,10 @@
   "overrides": {
     "request": {
       "form-data": "^2.5.4"
-    }
+    },
+    "js-yaml": "4.3.1",
+    "uuid": "11.1.1",
+    "qs": "6.14.1",
```
Comment (XyneSpaces) [inline packages/kata-sdk/package.json:26]: This forces the `uuid@3.4.0` pulled in transitively by `request` / `@kubernetes/client-node` up to v11. `request` calls uuid via the removed callable default export, so k8s HTTP calls can break at runtime with `uuid is not a function`. Verify the k8s client path actually works after this override (a real API call, not just install), or drop the override for uuid here.

File: apps/xyne-claw/package.json [hunk 3]
```diff
@@ -15,7 +15,7 @@
   "dependencies": {
     "@earendil-works/pi-agent-core": "0.75.5",
     "@earendil-works/pi-ai": "0.75.5",
-    "@earendil-works/pi-coding-agent": "0.75.5",
+    "@earendil-works/pi-coding-agent": "^0.80.3",
```
Comment (XyneSpaces) [inline apps/xyne-claw/package.json:18]: `pi-coding-agent` is bumped to `^0.80.3` while its sibling `pi-agent-core` and `pi-ai` stay pinned at exact `0.75.5`. This suite is versioned together; a 5-minor divergence risks runtime incompatibility, and switching this one to a caret range breaks the exact-pin discipline of its peers. Either bump all three together to matched versions with exact pins, or justify why only this one moved — and note this doesn't look like a Dependabot vuln fix, so it may be out of scope for this ticket.

File: apps/electron/package.json [hunk 4]
```diff
@@ -64,7 +64,7 @@
     "concurrently": "^9.2.1",
     "cpy-cli": "^7.0.0",
     "cross-env": "^10.1.0",
-    "electron": "^33.3.1",
+    "electron": "^39.8.10",
```
Comment (XyneSpaces) [inline apps/electron/package.json:67]: Electron `^33` → `^39` is a six-major jump (new Chromium + Node, breaking main/renderer API changes). The PR's 'Electron desktop' testing box is unchecked and there's no POT. Actually launch and smoke-test the desktop build before merge; installing cleanly is not evidence it runs.

File: apps/dashboard/package.json [hunk 5]
```diff
@@ -239,7 +239,7 @@
     "@typescript-eslint/parser": "^7.0.0",
     "@vitejs/plugin-react": "^5.0.2",
     "autoprefixer": "^10.4.21",
-    "electron": "^33.3.1",
+    "electron": "^39.8.10",
```
Comment (XyneSpaces) [inline apps/dashboard/package.json:242]: Same six-major Electron bump (`^33` → `^39`) as apps/electron. Keep these two in lockstep and validate the packaged desktop app, not just `pnpm install`.

File: apps/backend/package.json [hunk 6]
```diff
@@ -138,7 +138,7 @@
     "partial-json": "^0.1.7",
     "pdf-lib": "^1.17.1",
     "pdf-parse": "^2.4.5",
-    "pdfjs-dist": "^5.5.207",
+    "pdfjs-dist": "^6.2.108",
```
Comment (XyneSpaces) [inline apps/backend/package.json:141]: `pdfjs-dist` `^5` → `^6` is a major bump. The code fix in PdfJsStrategy.ts (`loadingTask.destroy()`) is correct and `loadingTask` is in scope, but confirm the `pdfjs-dist/legacy/build/pdf.mjs` entry path still resolves in v6 and that text extraction output is unchanged — attach a parse smoke-test / POT.

File: package.json [hunk 7]
```diff
@@ -221,77 +220,96 @@
...
       "form-data@2": "2.5.6",
       "form-data@4": "4.0.6",
       "handlebars@4": "4.7.9",
       "happy-dom": "20.8.9",
-      "hono": "4.12.30",
+      "hono": "4.12.34",
       "immutable": "4.3.9",
-      "js-yaml@3": "3.15.0",
-      "js-yaml@4": "4.3.0",
+      "js-yaml@3": "3.15.1",
```
Comment (XyneSpaces) [inline package.json:264]: `nodemailer` `8.0.11` → `9.0.1` is a major override bump. Confirm outbound email (transport config / auth) still works end-to-end before merge.

**Review by devesh-juspay — APPROVED** [review]

**Review by sakthin-juspay — APPROVED** [review]

**Review by KushagarGoel1 — APPROVED** [review]
